File: validAnagram/validAnagram.js
/**
 * IOCE
 * Input: two strings, s and t
 * Output: boolean
 * Contraints: n/a
 * Edge Cases: n/a
 *
 * Strategy
 * check length,
 * sort words
 * iterate through words and compare values
 *
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

const isAnagram = (s, t) => {
  if (s.length !== t.length) { return false; }

  s = s.split('').sort();
  t = t.split('').sort();

  for (let i = 0; i < s.length; i++) {
    if (s[i] !== t[i]) { return false; }
  }

  return true;
};

let s1 = 'anagram';
let t1 = 'nagaram';
console.log(isAnagram(s1, t1), 'should be true');

let s2 = 'rat';
let t2 = 'car';
console.log(isAnagram(s2, t2), 'should be false');

File: validAnagram/valid_anagram.py
def is_anagram(s, t):

  if len(s) != len(t):
    return False

  countS, countT = {}, {}

  for i in range(len(s)):
    countS[s[i]] = countS.get(s[i], 0) + 1
    countT[t[i]] = countT.get(t[i], 0) + 1

  for key in countS:
    if countS[key] != countT.get(key, 0):
      return False

  return True

s1 = 'anagram'
t1 = 'nagaram'
print(str(is_anagram(s1, t1)), 'should be true')

s2 = 'rat'
t2 = 'car'
print(str(is_anagram(s2, t2)), 'should be false')